pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -247,13 +247,14 @@ func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env
 }
 
 func (r *Builder) mapMacStaticIps(vm *model.VM) (ipMap string, err error) {
-	if !isWindows(vm) {
-		return "", nil
-	}
+	// on windows machines we check if the interface origin is manual
+	// on linux we collect all networks.
+	isWindowsFlag := isWindows(vm)
+
 	configurations := []string{}
 	gatewaySet := false
 	for _, guestNetwork := range vm.GuestNetworks {
-		if guestNetwork.Origin == string(types.NetIpConfigInfoIpAddressOriginManual) {
+		if !isWindowsFlag || guestNetwork.Origin == string(types.NetIpConfigInfoIpAddressOriginManual) {
```

Comment:
why we need to save it to var and not simply run `!isWindows(vm) || ....`?

Reply:
hmm, fixing :+1:

Reply:
simplified the for loop, by calling `isWindows(vm)` inside the loop

Reply:
oyy, it's calling a function inside a loop, vs using one variable ... reverting the fix ...